### PR TITLE
Headless compilation.

### PR DESCRIPTION
**Pull request by o01eg**
Splitted part of #1660. Adds configure option to build only server and AI binaries and some fixes for GUI-less environment.
Looks like it usable with `--external-server-address` client's option.

### CODE REVIEW

**Comment by geoffthemedio** [comment]
I suspect @adrianbroher would have some comments / concerns?

What are the "fixes for GUI-less environment"?

**Comment by o01eg** [comment]
It's for those systems which haven't any graphic libraries installed.

**Comment by geoffthemedio** [comment]
What, not for what.

**Comment by o01eg** [comment]
There is a second commit which removes `"../UI/LinkText.h"` from server dependencies.

**Comment by geoffthemedio** [comment]
Is that just removing an unnecessary header include? Presumably that's not specific for a GUI-less environment.

**Comment by o01eg** [comment]
`UI/LinkText.h` requires `glew` include files.

**Comment by geoffthemedio** [comment]
Sure, but does removing that include break anything, or is it entirely unnecessary regardless of whether the human client is also being built? If removing it means the server will not be fully functional if a human graphical client connects to it, that's also a problem.

**Comment by adrianbroher** [comment]
TL;DR: LGTM

> I suspect @adrianbroher would have some comments / concerns?

No objections,  the commit looks ok, albeit ugly.  But that's because noone is willing break up into the source into the components the software actually consists of.

> Sure, but does removing that include break anything, or is it entirely unnecessary regardless of whether the human client is also being built?

Why should it?  Any symbols provided by this include are not used.  If there were symbols used the compiler would complain about the missing declaration of those symbols.

**Comment by geoffthemedio** [comment]
I was just guessing there might have been a reason for the include being there. If it'd really unnecessary then there's no reason to keep it seemingly.
